### PR TITLE
Move shutdownfns, terminatefns and hammerfns out of separate goroutines

### PR DESCRIPTION
**Pull request by zeripath**
Coalesce the shutdownfns etc into a list of functions that get run at shutdown
rather then have them run at goroutines blocked on selects.

This may help reduce the background select/poll load in certain
configurations.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by zeripath** [comment]
Ref #15048 
Ref #15542

File: modules/graceful/manager_unix.go [hunk 1]
```diff
@@ -46,10 +54,10 @@ func newGracefulManager(ctx context.Context) *Manager {
 
 func (g *Manager) start(ctx context.Context) {
 	// Make channels
```
Comment (KN4CK3R) [inline modules/graceful/manager_unix.go]: Should be removed or changed.

File: modules/graceful/manager_windows.go [hunk 2]
```diff
@@ -58,11 +66,14 @@ func newGracefulManager(ctx context.Context) *Manager {
 }
 
 func (g *Manager) start() {
+
+	// Make channels
```
Comment (KN4CK3R) [inline modules/graceful/manager_windows.go]: same

File: modules/graceful/manager_windows.go [hunk 3]
```diff
@@ -58,11 +66,14 @@ func newGracefulManager(ctx context.Context) *Manager {
...
+	g.terminate, g.terminateCancel = context.WithCancel(ctx)
+	g.shutdown, g.shutdownCancel = context.WithCancel(ctx)
+	g.hammer, g.hammerCancel = context.WithCancel(ctx)
+	g.done, g.doneCancel = context.WithCancel(ctx)
+
 	// Make channels
```
Comment (KN4CK3R) [inline modules/graceful/manager_windows.go:75]: same

**Comment by zeripath** [comment]
I'm gonna close this to move its changes in to #15693

**Pull request closed without merging**